### DISCUSSION TITLE
Can you provide an easy installation guide for non-developers?

### DISCUSSION THREAD

**Discussion by amblerkr** [dubinc/dub#147] [Q&A]
Can you provide an easy installation guide for non-developers? I really like this one. 
I'm a marketer and I want to use this for a side-project.

It would be nice to have a document or video. 
If you provide it, I would like to translate it into Korean and let people know. 

This is a great project!

**Comment by fmerian** [comment]
Hi @amblerkr 👋 

Just to clarify -- do you need help setting up a custom domain on Dub?

Thanks!

**Reply by amblerkr** [reply]
yes!
https://github.com/steven-tey/dub/blob/main/.env.example
It will be easy to understand if there is a video.

It is difficult to set up a database `DATABASE_URL`.

**Reply by steven-tey** [reply]
Good point! I will be improving the setup flow + make a Loom video explaining the process soon! 🙏

**Reply by paulVu** [reply]
Waiting

**Reply by tobsowo** [reply]
> Good point! I will be improving the setup flow + make a Loom video explaining the process soon! 🙏 

I'll be interested in this.
Thank you.

**Comment by t1h0m** [comment]
I would love this as well. I'm not a programmer but generally decent-ish at installing / deploying open source code on my own hosting. I'm a little lost with the Configure Project variables I have to enter.

I'm currently finding the right prompts to get ChatGPT to help me understand and deploy, but if there's anything else I can refer to, that would be a huge help. 

amazing project btw!

**Reply by amblerkr** [reply]
I'm a non-developer too, and I'm comfortable with the digital landscape. With a little help, I can easily have a great product.

**Reply by t1h0m** [reply]
**Update:** seems like I was able to figure most of it out. Just stuck at one part: TEAM_ID_VERCEL

I'm following the [guide](https://vercel.com/guides/nextjs-multi-tenant-application#6.-add-custom-domains-with-the-vercel-api) but I didn't have a team ID since I don't have a team (on a solo free account on Vercel). Does this mean I won't be able to deploy?

**Reply by aiherrera** [reply]
Hey @t1h0m you can set your vercel user as your team ID. In the guide it states that if you are using a TEAM, which you aren't

![image](https://github.com/steven-tey/dub/assets/12240579/2f139e70-c87f-4d61-9e39-579db60dc9b0)

This would be the vercel URL:

`https://vercel.com/<TEAM_OR_USER_NAME>/<PROJECT_SLUG>`

**Comment by amblerkr** [comment]
I'm a marketer. I love dub, and I love it even more because it's open source. I'm not a developer either, but I'm familiar with digital environments and platforms. The service steven creates is fascinating. 

Shorten-url services have recently raised prices and have many restrictions.
In general, many shorten-url services appear temporarily and then go out of business.

I hope I can save the short url safely. I want to see statistics for all periods. Support for importing data exports. I don't want the link I created to die. no deadlink!

Dub is what I want, and I'd pay for it if it had a few improvements, and if it were sold by a place that sells source code, I'd buy it because it's such a cool service. It's pretty tiring to worry about it every time a service ends or a plan changes. I think there are a lot of people who just use it without thinking about it, and that includes marketers. They think that if they shorten the link and make it look good, that's it. Dub is fascinating. Hopefully I can host this and end this think. 

I already know and tried 'Yourls'. I will use this more often because I like dub.sh better.

**Comment by scesini** [comment]
Hello there,
Amazing project!
Maybe someone can help me, I tried to deploy on Vercel and I can see the:
"Note: one-click deployment is a bit broken at the moment – you'll need to change some of the hard-coded values in the codebase to get it working. We're working on fixing this."
Means, is not working for now?
I do setup the "Environment Variables" and still get multiple errors when I try to deploy like the next ones.
Does someone have a tip to properly deploy it?
Thank you in advance.-

```
Module not found: Can't resolve 'aws-crt' in '/vercel/path0/node_modules/@aws-sdk/util-user-agent-node/dist-cjs'
Module not found: Can't resolve 'react-native-sqlite-storage' in '/vercel/path0/node_modules/typeorm/driver/react-native'
Module not found: Can't resolve '@google-cloud/spanner' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve '@sap/hana-client' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'hdb-pool' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'mysql' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'oracledb' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'pg-native' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'pg-query-stream' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'ioredis' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'better-sqlite3' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'sqlite3' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'sql.js' in '/vercel/path0/node_modules/typeorm/platform'
Module not found: Can't resolve 'react-native-sqlite-storage' in '/vercel/path0/node_modules/typeorm/platform'
PrismaClientKnownRequestError: 
at pn.handleRequestError (/vercel/path0/node_modules/@prisma/client/runtime/library.js:176:6477)
at pn.handleAndLogRequestError (/vercel/path0/node_modules/@prisma/client/runtime/library.js:176:5907)
at pn.request (/vercel/path0/node_modules/@prisma/client/runtime/library.js:176:5786)
at async t._request (/vercel/path0/node_modules/@prisma/client/runtime/library.js:179:10484)
at async Object.generateStaticParams (/vercel/path0/.next/server/app/[domain]/page.js:738:63)
at async buildParams (/vercel/path0/node_modules/next/dist/build/utils.js:925:36)
at async /vercel/path0/node_modules/next/dist/build/utils.js:942:33
at async /vercel/path0/node_modules/next/dist/build/utils.js:1075:117
at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:105:20) {
code: 'P2021',
clientVersion: '4.13.0',
meta: { table: '(not available)' }
> Build error occurred
Error: Failed to collect page data for /[domain]
at /vercel/path0/node_modules/next/dist/build/utils.js:1164:15 {
type: 'Error'
error Command failed with exit code 1.
Error: Command "yarn run build" exited with 1
```
